llvm/include/llvm/Target/Target.td

```diff
@@ -1254,6 +1254,13 @@ def IMPLICIT_DEF : StandardPseudoInstruction {
   let isAsCheapAsAMove = true;
   let isMeta = true;
 }
+def INIT_UNDEF : StandardPseudoInstruction {
+  let OutOperandList = (outs unknown:$dst);
+  let InOperandList = (ins);
+  let AsmString = "";
+  let hasSideEffects = false;
+  let Size = 0;
```

Comment:
I initially also added `isMeta = true` here like IMPLICIT_DEF does, but this caused more (positive looking) changes in riscv, so I left it out again.

Disclaimer: I don't really know what I'm doing here... or what exactly the effect of isMeta is.

Reply:
I'm also wondering whether we could directly reuse IMPLICIT_DEF, as it's basically the same thing. We normally remove IMPLICIT_DEF and replace it with undef flags, but it would probably be fine to reintroduce it afterwards?

Reply:
For the record, I tried reusing IMPLICIT_DEF but based on the resulting RISCV test diffs at least that doesn't work (I saw some `vrgather.vi` with equal registers with that change).